pkg/distributor/distributor.go

```diff
@@ -209,14 +209,14 @@ func (d *Distributor) Push(ctx context.Context, req *logproto.PushRequest) (*log
 	validatedSamplesCount := 0
 
 	for _, stream := range req.Streams {
-		if err := d.validator.ValidateLabels(userID, stream.Labels); err != nil {
+		if err := d.validator.ValidateLabels(userID, stream); err != nil {
 			validationErr = err
 			continue
 		}
 
 		entries := make([]logproto.Entry, 0, len(stream.Entries))
 		for _, entry := range stream.Entries {
-			if err := d.validator.ValidateEntry(userID, entry); err != nil {
+			if err := d.validator.ValidateEntry(userID, stream.Labels, entry); err != nil {
```

Comment:
This seems a bit weird, did you swap the `Validate{Entry,Labels}`?

Reply:
not following?

Reply:
oops, problem between computer & chair

Reply:
labels are being passed in here to identify the stream in the error message, not for validation